src/index.rs

```diff
@@ -122,6 +127,7 @@ impl Index {
     /// assert_eq!(Index::Num(42).for_len_unchecked(30), 42);
     /// assert_eq!(Index::Next.for_len_unchecked(30), 30);
     /// ````
+    #[must_use]
```

Comment:
Any particular reason to make this one `must_use` and not the others?

Reply:
If the others are `Option` or `Result`, then they have it at the type level. 

I'll look closer when I'm at my computer.

Reply:
Ah right, that's probably what it is. 👍

Reply:
Yep, this was it. Great eye tho! I doubt I would ever have noticed that :)